crates/net/network/src/transactions/fetcher.rs

```diff
@@ -899,13 +910,7 @@ impl Hash for TxSizeMetadata {
 }
 
 impl TxSizeMetadata {
-    fn get_size(&self) -> Option<usize> {
-        if self.tx_encoded_len == 0 {
-            None
-        } else {
-            Some(self.tx_encoded_len)
-        }
-    }
```

Comment:
this is the code that is equivalent to `PackedOption`. so removing this OR `PackedOption`, that was the goal. not both. sorry if this was too advanced, hope you learned smthg new :)

Reply:
Oops! I missed this, let me write in a fix!

Reply:
My bad! I missed this! I've reverted it. I am a little new to Rust was wasn't the easiest, but lots of fun learning.

Reply:
ok looks really good now!

Reply:
Thank you, let me know if I need anything else to have it merged. Unsure what should I do about the warnings generated during build, though.

Reply:
before a PR is merged, CI should be all green, have you tried fixing merge conflicts?

Reply:
Just fixed the merge conflicts, but it seems like some methods and variables, since newly introduced, are unused. What is best steps for dealing with this? I can add a few tests too.